code/week02/src/Week02/Homework2.hs

```diff
@@ -43,23 +43,28 @@ PlutusTx.unstableMakeIsData ''MyRedeemer
 {-# INLINABLE mkValidator #-}
 -- This should validate if and only if the two Booleans in the redeemer are equal!
 mkValidator :: () -> MyRedeemer -> ValidatorCtx -> Bool
-mkValidator _ _ _ = True -- FIX ME!
+mkValidator () MyRedeemer{ flag1=f1, flag2=f2 } _ = traceIfFalse "wrong redeemer" $ f1 == f2
```

Comment:
Alternatively:
```Haskell
MyRedeemer{ .. } = flag1 == flag2
```

Reply:
Alternatively:

```
mkValidator () (MyRedeemer f1 f2) _ = traceIfFalse "wrong redeemer" $ f1 == f2
```

Even though `MyRedeemer` is a record-style data (with named fields) you can still construct/deconstruct it without those field names.

Reply:
@dino-, I assume this is because a record is syntactic sugar for a tuple in haskell?

Reply:
Alternatively:

```
mkValidator () r _ = traceIfFalse "wrong redeemer" $ flag1 r == flag2 r
```

The field names of a record data are automatically "getters". So `flag1 :: MyRedeemer -> Bool` and `flag2 :: MyRedeemer -> Bool`